### PR TITLE
Use Arguments in launch files and enable their reuse. 

### PR DESCRIPTION
**Pull request by destogl**
- Also add parameters for starting FakeSystem to demonstrate its functionality.

- [x] Add documentation about this Example and integrate into control.ros.org.

### CODE REVIEW

**Review by bmagyar — COMMENTED** [review]
happy with the changes once there's some examples

**Comment by olivier-stasse** [comment]
@destogl Do you need help on this pull request ?
I guess when it will be pushed this will invalidate/have a strong impact on other PRs.

**Comment by destogl** [comment]
@olivier-stasse would also like to get review from you.

**Comment by destogl** [comment]
Again some ros2 issues on the CI...

**Comment by olivier-stasse** [comment]
> @olivier-stasse would also like to get review from you.

Dear @destogl I was just reporting the constraints from the interface: it was saying that one reviewer with write access should validate the PR.

You, @bmagyar and @Karsten1987 are doing an incredible job, I was just trying to avoid having PR piling up without proper grounds. 

Thanks again for your time and dedication.

**Comment by olivier-stasse** [comment]
> Again some ros2 issues on the CI...

So we have to wait that the package number is updated (I checked the number of the package required and indeed it does not correspond) ? Or is it a transient situation with the buildfarm ?

File: README.md
Comment (jordan-palacios) [inline README.md:155]: I think you meant this:

```suggestion
- Launch file: rrbot_system_multi_interface.launch.py
```

File: README.md
Comment (jordan-palacios) [inline README.md:184]: For this demo the default rviz configuration file did not set up correctly the fixed frame (I'm guessing it's `odom`) nor did it showed the Robot Model.

After fixing this and sending the velocity command the demo runs fine.

One thing I noticed is the velocity being correctly updated in the `/joint_states` while the wheel positions is fixed to zero. 

```
header:
  stamp:
    sec: 1623246367
    nanosec: 753984445
  frame_id: ''
name:
- right_wheel_joint
- left_wheel_joint
position:
- 0.0
- 0.0
velocity:
- 50.0
- 43.33333333333333
effort:
- .nan
- .nan
```

**Review by jordan-palacios — COMMENTED** [review]
Demos work fine for me aside for a few minor details.

Current Example enumeration goes like this: 1 -> 1 -> 2 -> 10 :D

File: README.md
Comment (olivier-stasse) [reply README.md:184]: Thanks @jordan-palacios I will try to propose a fix by tomorrow morning.

File: README.md
Comment (mahaarbo) [inline README.md:30]:
```suggestion
* Creating a `*HardwareInterface` for a System, Sensor, and Actuator.
```

File: README.md
Comment (mahaarbo) [inline README.md:35]:
```suggestion
* Implementing a controller switching strategy for a robot.
```

File: README.md
Comment (mahaarbo) [inline README.md:42]:
```suggestion
* There are now three categories of hardware interface components: *Sensor*, *Actuator*, and *System*.
```
Shouldn't this be hardware interface components? Later we talk of the different hardware interfaces types that can be defined, and it may be a bit confusing.

File: README.md
Comment (mahaarbo) [inline README.md:45]:
```suggestion
  All components are used as plugins and therefore exported using `PLUGINLIB_EXPORT_CLASS` macro.
```

File: README.md
Comment (mahaarbo) [inline README.md:59]:
```suggestion
**NOTE**: `ros2_control` and `ros2_controllers` packages are released for foxy and can be installed using a package manager.
```

File: README.md
Comment (mahaarbo) [inline README.md:91]:
```suggestion
*RRBot*, or ''Revolute-Revolute Manipulator Robot'', is a simple 3-linkage, 2-joint arm that we will use to demonstrate various features. It is essentially a double inverted pendulum and demonstrates some fun control concepts within a simulator and was originally introduced for Gazebo tutorials.
```

File: README.md
Comment (mahaarbo) [inline README.md:96]:
```suggestion
1. To start an example open a terminal, source your ROS2-workspace and execute a launch file with:
```

File: README.md
Comment (mahaarbo) [inline README.md:101]:
```suggestion
2. To check if the hardware interface loaded properly, open another terminal and execute:
```

File: README.md
Comment (mahaarbo) [inline README.md:115]:
```suggestion
3. Check which controllers are running using:
```

File: README.md
Comment (mahaarbo) [inline README.md:125]:
```suggestion
4. Check [Controllers and moving hardware](#Controllers-and-moving-hardware) section to move *RRBot*.
```

File: README.md
Comment (mahaarbo) [inline README.md:213]:
```suggestion
You should now see an orange box circling in `rviz2`.
```

File: README.md
Comment (mahaarbo) [inline README.md:216]:
```suggestion
## Controllers and moving hardware
```

File: README.md
Comment (mahaarbo) [inline README.md:261]:
```suggestion
   You should get the response:
```

File: README.md
Comment (mahaarbo) [inline README.md:280]:
```suggestion
3. Send a command to the controller, either:
```

File: README.md
Comment (mahaarbo) [inline README.md:288]:
```suggestion
   B. Or you can start a demo node which sends two goals every 5 seconds in a loop:
```

File: doc/index.rst
Comment (mahaarbo) [inline doc/index.rst:11]: What is meant by this? Should this be
```suggestion
**Note:** the following list is comprehensive.
```
Or that it is not comprehensive, as the ``ros2_control_demo_bringup`` etc are not mentioned?

**Review by mahaarbo — COMMENTED** [review]
I've made some minor comments on the documentation, hope you don't mind!

File: README.md
Comment (Karsten1987) [reply README.md:42]: I'd leave out the word 'interfaces' completely. We've been using `components` everywhere else in the code base.

File: ros2_control_demo_bringup/CMakeLists.txt
Comment (Karsten1987) [inline ros2_control_demo_bringup/CMakeLists.txt:12]:
```suggestion
```

File: ros2_control_demo_description/CMakeLists.txt
Comment (Karsten1987) [inline ros2_control_demo_description/CMakeLists.txt:9]:
```suggestion
```

**Comment by olivier-stasse** [comment]
PR #104 adresses the reviews. Thanks to all.

**Comment by destogl** [comment]
> Dear @destogl I was just reporting the constraints from the interface: it was saying that one reviewer with write access should validate the PR.

This setup ensured the quality of packages in general. Nevertheless, we value every review because this guarantees more eyes and even higher quality and sensibility of an implementation :)

File: README.md
Comment (destogl) [reply README.md:33]: I will leave this for the follow up PR. see #105

File: README.md
Comment (destogl) [inline README.md:42]:
```suggestion
* There are now three categories of hardware components: *Sensor*, *Actuator*, and *System*.
```

File: README.md
Comment (destogl) [inline README.md:91]:
```suggestion
*RRBot*, or ''Revolute-Revolute Manipulator Robot'', is a simple 3-linkage, 2-joint arm that we will use to demonstrate various features.
It is essentially a double inverted pendulum and demonstrates some fun control concepts within a simulator and was originally introduced for Gazebo tutorials.
```

**Comment by destogl** [comment]
Thank you all for the input! Especially to:

- @jordan-palacios - for being detailed in testing
- @mahaarbo - for reading the README carefully
- @olivier-stasse - for the support to quickly resolve major issues.

It's really great to be a part of such a community :)

**Review by olivier-stasse — APPROVED** [review]

**Comment by olivier-stasse** [comment]
Dear @destogl and all,
Do you think that it will be possible to merge this PR in master or is there other things to do ?
Issue #75 could be solved in a different PR. This one is already _really_ big. I believe that `add-rrbot- corrections` is already acting like a devel branch.

**Review by Karsten1987 — APPROVED** [review]
I agree with Olivier. Let's get this one in and follow up with other smaller increments in case it's still not fully accurate.

**Review by bmagyar — APPROVED** [review]

**Comment by bmagyar** [comment]
Phew, this was quite a marathon to review! :D merging...

**Comment by destogl** [comment]
@Karsten1987  @bmagyar thanks for final approvals!

**Comment by Karsten1987** [comment]
@destogl did you ever test the gazebo demo of this PR?
I get a segfault when trying to launch the gazebo rrbot on OSX as well as Linux:

```
[gzserver-1] [Err] [Model.cc:1137] Exception occured in the Load function of plugin with name[gazebo_ros2_control] and filename[libgazebo_ros2_control.so]. This plugin will not run.
[ros2_control_node-3] [INFO] [1623786805.821751090] [RRBotSystemPositionOnlyHardware]: 0.0 seconds left...
[ros2_control_node-3] [INFO] [1623786805.821794391] [RRBotSystemPositionOnlyHardware]: System Successfully started!
[ERROR] [gzserver-1]: process has died [pid 32476, exit code -11, cmd 'gzserver       --verbose                                                                -s libgazebo_ros_init.so   -s libgazebo_ros_factory.so   -s libgazebo_ros_force_system.so       '].
[gzserver-1]
```

I guess that's really on me though, I should have tested it correctly before approving.

**Comment by olivier-stasse** [comment]
@Karsten1987 @destogl  my bad too. 

I have a fix for starting Gazebo here https://github.com/olivier-stasse/ros2_control_demos/commit/47129957d2d32d040f833e6c2373daabc4dcb479 

I will try to test the controller probably having in mind the fixes in PR #103 .

**Pull request merged**
